### ISSUE TITLE
Community information does not load for communities with apostrophe in name

### ISSUE THREAD

**Issue by lindsayking** [Esri/site-selector#207]
Related to https://github.com/Esri/site-selector/issues/159

When the community drop-down is used to search for a community, the application spins indefinitely is a search is performed on a community with an apostrophe in the name.

Steps to repro:
1) Launch http://50.18.115.76/SiteSelectorSG and click OK on the splash screen
2) Click on the Communities tab
3) Turn on the radio button to "Search Communities in ..."
4) Select either Prince George's, Queen Anne's, or Saint Mary's from the list
5) Note the application spins indefinitely.  The map does not update and no geoenrichment information loads.

This behavior only appears to manifest if there is an apostrophe in the name.

**Comment by CTLocalGovTeam** [comment]
@lindsayking, @LindsayThomas, @NikkiGolding

Text that contain a single quotation mark (apostrophe) need to be escaped when used for querying. The solution is to use an additional quotation mark to execute the query.
For e.g., 
`NAME = 'Prince George's'   //incorrect`
`NAME = 'Prince George''s'   //correct`

We identified two places in the app where this needs to be handled. They are:
1. Unified search feature in Buildings and Sites tab
2. Filter feature (regular and additional filter options in configuration)

We tested this with other special characters and did not encounter any issues that need to be handled specifically like the apostrophe issue.

Also, the app spins indefinitely because the busy screen was not cleared on error. We found a couple of places where this needs to be handled to ensure that the busy screen is cleared on error. This is a minor code change and will help to avoid showstoppers when the busy screen is not cleared.

**Comment by lindsayking** [comment]
@NikkiGolding @LindsayThomas 

Tested the v1.9.2 fix for this issue.

* Community names containing apostrophes no longer cause the app the hang.  The app behaves correctly and sharing is successful with these community names.  Names that contain spaces remain fine (#200 remains fixed).
* Tested with unified searches containing apostrophes.  These searches resulted in multiple error messages and then no search results in earlier versions of the app.  Fixed now with v1.9.2.
* Tested filtering on a field value containing an apostrophe in RegularFilterOprtionFields for both Buildings and Sites. 
```
RegularFilterOptionFields: [
								{
									DisplayText: "Prince George's County",
									FieldName: "SITECOUNTY",
									FieldValue: "Prince George's",
								}
                            ],
```
Failed in earlier versions of the app.  The results were not filtered and an error displayed in the console:
![image](https://cloud.githubusercontent.com/assets/4316336/12023492/e45b6c20-ad4e-11e5-9621-42184e6f63fd.png)

Using the same RegularFilterOptionFields settings with v1.9.2 now successfully displays the correctly filtered results with no errors displaying in the console.
* Tested with filtering on field values containing apostrophes in AdditionalFilterOptions for both Buildings and Sites:
```
AdditionalFilterOptions: {
                                Enabled: true,
                                FilterFieldName: "SITECOUNTY",
                                FilterOptions: [
                                    {
                                        DisplayText: "Howard",
                                        FieldValue: "Howard"
                                    }, {
                                        DisplayText: "Prince George's",
                                        FieldValue: "Prince George's"
                                    }, {
                                        DisplayText: "Saint Mary's",
                                        FieldValue: "Saint Mary's"
                                    }
                                ]
                            }
```
Failed in earlier versions of the app (results did not filter and error appeared in console).  Fixed with v.1.9.2.

Closing this issue.  All test cases now working correctly with v1.9.2.
